### PR TITLE
Advance receipt adjustment implemented (Ledger, invoice, sales invoice )

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 1]
```diff
@@ -723,98 +721,98 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
                     </div>
 
                     <!-- Advance receipts adjusted Invoice -->
-<!--                    <div class="clearfix adjust-receipt font-14"-->
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:726]: We will be use all Commented code  in next version

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 2]
```diff
@@ -1479,4 +1484,47 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
+    /**
+     * To check adjusted advance amount is more  than advance receipt/invoice
+     *
+     * @param {number} totalAmount Total compound amount
+     * @memberof UpdateLedgerEntryPanelComponent
+     */
+    public checkAdjustedAmountExceed(totalAmount: number): void {
+        if (Number(this.vm.compoundTotal) < Number(totalAmount)) {
+            this.isAdjustedAmountExcess = true;
+            this.adjustedExcessAmount = totalAmount - this.vm.compoundTotal;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1497]: Wrap totalAmount and this.vm.compoundTotal by Number

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss [hunk 3]
```diff
@@ -420,54 +426,63 @@ button.btn.btn-filter span.icon-cross {
...
 }
 
+span.icon-export-icon {
+    color: $gray;
+}
+
+.invoice-footer {
+    display: flex;
+    padding: 20px;
+    background: #fff;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss:463]: Use color variables

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss [hunk 4]
```diff
@@ -559,21 +567,34 @@ div.remove-capitalize {
     .desktop-view-btn {
         display: none;
     }
+    .date-advanceSearch {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss:570]: class name in camelCase

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss [hunk 5]
```diff
@@ -583,17 +604,30 @@ div.remove-capitalize {
...
-    .invoiceFooter {
+    .invoice-footer {
         display: flex;
         padding: 10px 0;
     }
+    .full-width-on-mobile {
+        width: 100%;
+        margin: 5px 0;
+    }
+    .date-advanceSearch {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss:628]: class name in camelCase

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 6]
```diff
@@ -280,7 +281,9 @@ export class InvoicePreviewComponent implements OnInit, OnChanges, OnDestroy {
         this.voucherDetails$ = this.store.pipe(select(s => s.receipt.voucher), takeUntil(this.destroyed$));
         //this._invoiceService.getTotalAndBalDue();
     }
-
+    openModal(template: TemplateRef<any>) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts:284]: Do add JS DOC and return type

File: apps/web-giddh/src/app/invoice/templates/edit-template/modals/confirmation-modal/confirmation.modal.component.html [hunk 7]
```diff
@@ -5,7 +5,14 @@ <h3>Confirmation</h3>
     </div>
     <div class="modal-body clearfix" id="export-body">
         <form name="newRole" novalidate class="" autocomplete="off">
-            <h3 [innerHTML]="message"></h3>
+            <h3 *ngIf="!isMessageTextOrHeading" [innerHTML]="message"></h3>
+            <ng-container *ngIf="isMessageTextOrHeading">
+                <div class="modal_wrap mrB2">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/templates/edit-template/modals/confirmation-modal/confirmation.modal.component.html:10]: Class names in underscore

File: apps/web-giddh/src/app/invoice/templates/edit-template/modals/confirmation-modal/confirmation.modal.component.ts [hunk 8]
```diff
@@ -1,16 +1,21 @@
...
     templateUrl: './confirmation.modal.component.html'
 })
 
-export class DeleteTemplateConfirmationModelComponent {
+export class DeleteTemplateConfirmationModelComponent implements OnChanges {
 
     @Input() public message: string;
     @Input() public flag: string;
     @Output() public closeModelEvent: EventEmitter<any> = new EventEmitter(null);
+    public messageBody: string = '';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/templates/edit-template/modals/confirmation-modal/confirmation.modal.component.ts:14]: JS doc

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 9]
```diff
@@ -2040,9 +2050,9 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
         if (isNaN(count)) {
             count = 0;
         }
-        // if (!depositAmount) {
-        //     this.depositAmount = this.invFormData.voucherDetails.deposit;
-        // }
+        if (!depositAmount) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:2053]: Remove this code @rohit-walkover

**Pull request merged**